GameSync.Front/src/app/features/game-detail/game-detail.component.ts

```diff
@@ -29,46 +29,48 @@ export class GameDetailComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    this.refresh();
-  }
 
-  refresh() {
-    this.gameResult$ = this.route.url.pipe(
-      switchMap(params => {
-        this.isCustom = params[0].path === 'custom';
+    this.route.url.pipe(
+      switchMap(segments => {
+        this.isCustom = segments[0].path === 'custom';
         if (this.isCustom) {
           return this.gamesService
-            .getCustomGameDetail(params[1].path)
-            .pipe(map(this.wrapToResult));
+            .getCustomGameDetail(segments[1].path)
+            .pipe(map(game => ({inCollection: false, game})));
         }
+        return this.gamesService.getGameDetail(segments[0].path);
+      }))
```

Comment:
Je ne sais pas si c'est mieux, c'est une proposition:

```typescript
this.route.url.pipe(
   switchMap(segment =>
      iif(() => segments[0].path === 'custom', //predicate
                   this.gamesService.getCustomGameDetail(segments[1].path //if true
                      .pipe(map(game => ({inCollection: false, game})),
                   this.gamesService.getGameDetail(segments[0].path))) //if false
      )
   )
).subscribe(...)
```

Je le fais à la volée. Tu peux regarder l'utilisation de l'opérateur iif() ici: 
https://www.learnrxjs.io/learn-rxjs/operators/conditional/iif

Reply:
J'aime l'idée. par contre y'a plus l'assignation "this.isCustom = [...]" non ? Le front marcherait pas sans. Peut-être qu'il faudrait faire un tap() avant pour faire cet effet de bord mais je pense pas que ce soit super propre.

Reply:
Oui tu as raison. Va pour un `tap()`